### PR TITLE
fix: Refactor environment data structure

### PR DESCRIPTION
**Pull request by itleigns**
## Context
As described in #77 and #80, sd-local fails to decode or encode environment variables in `screwdriver.yaml` in some cases because we do not use library in UnmarshalJSON nor MarshalJSON [here](https://github.com/screwdriver-cd/sd-local/blame/44ba559986823c61776715d7abd453c9c5edaacd/screwdriver/screwdriver.go#L54-L75).
Also, the UnmarshalJSON of `EnvVar` is not the inverse function of the MarshalJSON, which is confusing.

## Objective
- We use [go-ordered-json](https://pkg.go.dev/gitlab.com/c0b/go-ordered-json) to implement UnmarshalJSON.
- We use raw `[]map[string]string` as the data structure for the buildEntry environments and use `[]map[string]string`'s own MarshalJSON.

## References
#73
#74
#77
#80

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: launch/docker.go [hunk 1]
```diff
@@ -314,3 +314,16 @@ func signum(sig os.Signal) int {
...
+
+// GetEnv gets the newest value whose key is key
+func GetEnv(en []map[string]string, key string) string {
+	s := ""
+	for _, e := range en {
+		for k, v := range e {
+			if k == key {
+				s = v
+			}
+		}
```
Comment (topia) [inline launch/docker.go:326]:
```suggestion
		if v, ok := e[key]; ok {
			s = v
		}
```

**Review by wahapo — APPROVED** [review]

**Review by esuc4se — DISMISSED** [review]

File: screwdriver/screwdriver.go [hunk 2]
```diff
@@ -44,70 +43,42 @@ type Step struct {
...
-	})
+	return nil
 }
 
-// Merge en2 to en
-func (en *EnvVar) Merge(en2 EnvVar) {
-	for _, e := range en2 {
-		en.Set(e.Key, e.Value)
+// Merge associative array to EnvVars
+func (en *EnvVars) Merge(en2 map[string]string) {
```
Comment (yoshwata) [inline screwdriver/screwdriver.go:71]: This function may be better named `AppendAll` than `Merge`

The name merge is reminiscent of the following behavior:
```
o1 = { a: aaa, b: bbb}
o2 = { b: foo, c: bar}

o3 = Merge(a, b)
// o3: { a: aaa, b: foo, c: bar}
```

Comment (itleigns) [reply screwdriver/screwdriver.go]: I fixed.

File: launch/docker.go [hunk 3]
```diff
@@ -314,3 +314,14 @@ func signum(sig os.Signal) int {
 		return -1
 	}
 }
+
+// GetEnv gets the newest value whose key is key
```
Comment (yoshwata) [inline launch/docker.go:318]:
```suggestion
// GetEnv returns the newest value corresponding to the key
```

**Review by yoshwata — APPROVED** [review]

**Review by tkyi — APPROVED** [review]

**Pull request merged**